go/mysql/conn.go

```diff
@@ -1030,6 +1050,7 @@ func (c *Conn) handleNextCommand(handler Handler) error {
 			}
 		}
 	case ComPrepare:
+		log.Info("Received COM_PREPARE")
```

Comment:
The server connection has become quite a bit more stateful now, because it can be in this cursor-exists state with the query running in the background. I think almost all of these cases need to check for `conn.cs != nil` and do the right thing...I'm not exactly sure what the right thing is for each of them. Maybe connection out of sync error and close the connection?

Seems like ComStmtReset should definitely cancel the query, ensure the query is canceled by selecting on its done channel, and only return success after that's done.

conn.StatusFlags will need to be handled carefully across anywhere the `conn.cs` can go from `!= nil` to `== nil`, since we need to make sure CursorExists is getting cleared.